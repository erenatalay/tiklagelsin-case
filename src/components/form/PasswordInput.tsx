import React, { FC } from 'react'
import { StyleSheet, View, useColorScheme } from 'react-native'
import { TextInput, TextInputProps, DefaultTheme } from 'react-native-paper';
import ErrorText from './ErrorText';
import Colors from '../../constant/Colors';
import Ionicons from '@expo/vector-icons/Ionicons';

interface InputProps extends TextInputProps {
    name: string
    touched?: boolean
    changeText: (input: string, text: string) => void
    blur: (input: string) => void
    errors?: string
}
const PasswordInput: FC<InputProps> = (props) => {
    const [isSecureTextEntry, setSecureTextEntry] = React.useState<boolean>(true);
    const { name, placeholder, left, keyboardType, value, changeText, blur, touched, secureTextEntry = false, errors } = props;
    const color = useColorScheme() ?? 'light';
    const theme = {
        ...DefaultTheme,
        colors: {
            ...DefaultTheme.colors,
            background: "white",
            text: "blue",
        },
        ...{ roundness: 30 },

    };
    return (
        <View style={styles.container}>
            <TextInput
                theme={theme}
                left={left}
                value={value}
                onBlur={() => blur(name)}
                keyboardType={keyboardType}
                placeholder={placeholder}
                placeholderTextColor={Colors[color].textInputPlaceholder}
                activeOutlineColor={Colors[color].textInputBackground}
                autoComplete='off'
                secureTextEntry={isSecureTextEntry}
                style={[styles.textInput, { height: props.multiline ? 60 : 30 }]}
                mode={"outlined"}
                right={<TextInput.Icon
                    onPress={() => setSecureTextEntry(!isSecureTextEntry) }
                    icon={isSecureTextEntry ? "eye" : "eye-off"} />
                }
                onChangeText={(text: string) => changeText(name, text)}
                {...props}
            />
            <ErrorText error={errors} touched={touched} />
        </View>
    )
}
const styles = StyleSheet.create({
    container: {
        marginVertical: 10
    },
    textInput: {
        paddingVertical: 15,
        borderRadius: 64,
        height: 30,
        justifyContent: "center",
    },
})
export default PasswordInput;