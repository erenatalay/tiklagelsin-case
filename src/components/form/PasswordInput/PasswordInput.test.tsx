import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import PasswordInput from './PasswordInput';
describe('Input component tests', () => {
    it('is rendered correctly', () => {
        expect(<PasswordInput
            name="inputName"
            placeholder="Enter text"
            changeText={() => null}
            blur={() => null}

        />
        ).toMatchSnapshot();
    });

    it('should trigger Password when typed', () => {
        const mockSearch = jest.fn();
        const wrapper = render(
            <PasswordInput
                name="inputName"
                placeholder="Enter text"
                changeText={mockSearch}
                blur={() => null}

            />
        );
        const input = wrapper.getByTestId('password-input');
        fireEvent(input, 'onChangeText', 'test');
        expect(mockSearch).toHaveBeenCalled();
    });
});

