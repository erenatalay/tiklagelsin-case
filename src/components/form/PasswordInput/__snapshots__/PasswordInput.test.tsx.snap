// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests is rendered correctly 1`] = `
<PasswordInput
  blur={[Function]}
  changeText={[Function]}
  name="inputName"
  placeholder="Enter text"
/>
`;
